engine/colliderbox.h

```diff
@@ -1,13 +1,15 @@
 #pragma once
 
+#include "component.h"
+
 #include <glm/glm.hpp>
 
 #include <array>
 
 namespace pong
 {
 
-class ColliderBox
+class ColliderBox : public Component<ColliderBox>
```

Comment:
Ooh, the CRTP!! Literally was just reading a chapter of a book about it.

This was also how Roblox did static compile time polymorphism, so you're in good company "game engine" wise.